rest-api/api/participant_counts_over_time_api.py

```diff
@@ -0,0 +1,102 @@
+from flask import request
+from werkzeug.exceptions import BadRequest
+
+from api_util import HEALTHPRO
+from api_util import get_awardee_id_from_name
+from app_util import auth_required
+from dao.hpo_dao import HPODao
+from participant_enums import EnrollmentStatus
+
+DATE_FORMAT = '%Y-%m-%d'
+DAYS_LIMIT = 100  # provisional, per design doc
+
+
+class ParticipantCountsOverTimeApi(Resource):
+
+  @auth_required(HEALTHPRO)
+  def get(self):
+    self.hpo_dao = HPODao()
+
+    # TODO: After enrollment status is filterable,
+    # wire in 'organization', 'site', and 'enrollmentStatus'.
+    enrollment_status = request.args.get('enrollmentStatus')
+    awardee = request.args.get('awardee')
+    stratification = request.args.get('stratification')
+    start_date = request.args.get('startDate')
+    end_date = request.args.get('endDate')
+
+    params = {
+      'enrollment_statuses': enrollment_status,
+      'awardees': awardee,
+      'stratification': stratification,
+      'start_date': start_date,
+      'end_date': end_date
+    }
+
+    # Most parameters accepted by this API can have multiple, comma-delimited
+    # values.  Arrange them into lists.
+    for param in params:
+      value = params[param]
+      if param in ['start_date', 'end_date', 'stratification']:
+        params[param] = value.encode()
+        continue
+      if value is None:
+        params[param] = []
+      else:
+        params[param] = value.encode().split(',')
+
+    params = self.validate_params(params)
+
+
+  def validate_params(self, params):
+
+    start_date_str = params['start_date']
+    end_date_str = params['end_date']
+    enrollment_statuses = params['enrollment_statuses']
+    awardees = params['awardees']
+    stratification = params['stratification']
+
+    # Validate dates
+    if not start_date_str or not end_date_str:
+      raise BadRequest('Start date and end date should not be empty')
+    try:
+      start_date = datetime.datetime.strptime(start_date_str, DATE_FORMAT).date()
+    except ValueError:
+      raise BadRequest('Invalid start date: %s' % start_date_str)
+    try:
+      end_date = datetime.datetime.strptime(end_date_str, DATE_FORMAT).date()
+    except ValueError:
+      raise BadRequest('Invalid end date: %s' % end_date_str)
+    date_diff = abs((end_date - start_date).days)
+    if date_diff > DAYS_LIMIT:
+      raise BadRequest('Difference between start date and end date ' \
+                       'should not be greater than %s days' % DAYS_LIMIT)
+    params['start_date'] = start_date
+    params['end_date'] = end_date
+
+    # Validate awardees, get ID list
+    awardee_ids = []
+    for awardee in awardees:
+      awardee_id = get_awardee_id_from_name({'awardee': awardee}, self.hpo_dao)
+      if awardee_id == None:
+        raise BadRequest('Invalid awardee name: %s' % awardee)
+      awardee_ids.append(awardee_ids)
+    params['awardee_ids'] = awardee_ids
+
+    valid_enrollment_statuses = EnrollmentStatus.to_dict()
```

Comment:
1) Do we need to support multiple enrollment statuses in the filter? 

2) You should probably convert the enrollment statuses to enums here... something like:

try:
  params["enrollment_statuses"] = [EnrollmentStatus(val) for val in enrollment_statuses]
except ValueError:
  raise BadRequest('Invalid enrollment status: %s' % enrollment_statuses)

Reply:
> Do we need to support multiple enrollment statuses in the filter?

Yes, I think so.  I can envision cases where users would want to apply filters to _exclude_ participants in one enrollment status, which we'd presumably support for passing all enrollment statuses except that one.

> You should probably convert the enrollment statuses to enums here... something like

Good point, done, thanks.  I was using the approach from [metric_sets_api.py](https://github.com/all-of-us/raw-data-repository/blob/40a6a141c780d232feafd8b1b4d7605e29b44b45/rest-api/api/metric_sets_api.py#L26), but yours is more consistent with other validation in this class.

Reply:
Actually, @danqrodney,  if we support multiple enrollment statuses in the filter as I think is needed, then my original `EnrollmentStatus.to_dict()` seems preferable, because the `raise BadRequest('Invalid enrollment status: %s' % enrollment_statuses)` statement outputs the list of specified enrollment status filters, even those only one may be invalid.

Reply:
Meh. You don't normally expect the client to send you bad values; if it does it won't be that hard to work out which one it is. I'd stick with the change you made.